src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/FindReferences/FindReferencesEndpointTest.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Internal;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Microsoft.VisualStudio.Threading;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.FindAllReferences;
+
+public class FindReferencesEndpointTest : SingleServerDelegatingEndpointTestBase
+{
+    public FindReferencesEndpointTest(ITestOutputHelper testOutput) : base(testOutput)
+    {
+    }
+
+    [Fact]
+    public Task FindCSharpReferences()
+        => VerifyCSharpFindAllReferencesAsyncAsync("""
+
+                @{
+                    const string [|$$S|] = "";
```

Comment:
This is where my knowledge of the razor language is failing me a little. We don't support HTML contexts (so we don't need to test `<MyComponent>` for example). Are there some corner cases I'm missing? 

Also I think we should add support for Razor directives, but this just mirrors what was supported before.

Reply:
I think this is good. I have a test in a PR (skipped for now) for component attributes, which i'll unskip if this goes in first, and supporting components should be possible (actually, free) once #6887 is done.

Not sure what scenario makes sense for directives exactly, but feel free to log an issue. There are already a couple of issues logged for inject or inherit directives I think.